src/DTO/Model/HierarchyDTO.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace App\DTO\Model;
+
+
+/**
+ * @template T
+ */
+class HierarchyDTO
```

Comment:
if I understand it correctly you use this DTO to first build the tree with business entities to afterwards convert the business entity to the actual AnswersDTO. This can cause huge problems. If for some reason one entity is accidentally skipped and doesn't get converted to an actual DTO then the DB entity gets returned in the API. This is already bad but with doctrine this could cause that all referenced entities get fetched and also returned. In really bad cases this could theoretically cause the whole DB to be exposed through the API.

In short: Never have business / db entities in objects which have DTO in their name! You would expect DTOs to only contain save and sanitized data.